src/commands/Minion/alch.ts

```diff
@@ -26,11 +26,94 @@ const unlimitedFireRuneProviders = resolveItems([
 	'Tome of fire'
 ]);
 
+export function calculateAlchItemByTripDuration(params: {
+	user: KlasaUser;
+	duration: number;
+	items?: { item: Item; qty: number }[];
+	qtyLimit?: number;
+	calculateByRunesOwned?: boolean;
+}) {
+	let { items, duration, user, qtyLimit, calculateByRunesOwned } = params;
+	const userBank = user.bank().clone();
+
+	if (!items) items = [];
+
+	if (items.length === 0) {
```

Comment:
This should be able to be reworked with a map.

Something like
```ts
const favAlchs = user.getUserFavAlchs().map(fa => {
	if (userBank.amount(fa.id))
		return {
			item: fa,
			qty: qtyLimit
				? qtyLimit < userBank.amount(fa.id)
					? qtyLimit
					: userBank.amount(fa.id)
				: userBank.amount(fa.id)
		};
}).filter(fa => fa !== undefined);
const allItemsToAlch = items.length === 0 ? favAlchs : items;
```

Then use the allItemsToAlch array

Reply:
Why? The reason it is like that, is so we don't run that when the user just wants to alch an item. Why go over the user favs if he is trying to force alch something. This will only run if the user does `+alch` or when the user is doing alch agility.